code/modules/mapfluff/ruins/spaceruin_code/hauntedtradingpost.dm

```diff
@@ -245,7 +245,7 @@
 	//is this being used as part of the haunted trading post ruin? if true, will self destruct when boss dies
 	var/donk_ai_slave = FALSE
 	// machine that the trap inhabits
-	var/obj/structure/host_machine
+	var/obj/host_machine
```

Comment:
why just /obj

Reply:
i assumed the existing typehint wasn't completely wrong and went with the ancestor of `/obj/structure` and `/obj/machinery`, but it seems it literally can't be anything other than `/obj/machinery`